Automattic/wp-calypso
# State: Implement Post by Email into Jetpack Settings redux state

### Background

The email in Post by Email Jetpack functionality behaves a little differently from the rest of the Jetpack Settings. When retrieved from the settings endpoint, it simply provides the current email address, however when updating Jetpack Settings we can only send a `post_by_email_address: 'regenerate'` parameter, which regenerates the remote email address. This behavior requires a special treatment of that field type.

### Purpose

This PR implements the necessary Redux functionality into the Jetpack Settings redux tree, specifically:

* Adding a new action to trigger regenerating of the email: `regeneratePostByEmail()`.
* Updating reducer to handle expected changes of the email address in the settings tree.
* Adding a new `isRegeneratingPostByEmail` selector, used to indicate whether we're currently regenerating the email address.
* Omitting `post_by_email_address` on purpose when sanitizing settings before updating them.
* Tests for all of the above changes.

We'll need all of this to complete the Publishing Tools card - #10663. This PR is part of the Jetpack Settings in Calypso effort - #9171.

### Testing

* Checkout this branch
* Verify all Jetpack Settings test are passing: 
```
npm run test-client client/state/jetpack/settings/
```
